CDEFramework/Libraries/LegacyDictionarySupport/GlossaryAPIClient.cs

```diff
@@ -0,0 +1,152 @@
+            if (String.IsNullOrWhiteSpace(dictionary))
+            {
+                throw new ArgumentNullException("The dictionary is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(audience))
+            {
+                throw new ArgumentNullException("The audience is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(language))
+            {
+                throw new ArgumentNullException("The language is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentNullException("The ID is null or an empty string");
+            }
+
+            // Set up search param string: {dictionary}/{audience}/{language}/{id:long}
+            string[] searchParams = { dictionary, audience, language, id, useFallback.ToString() };
+            string searchParam = string.Format("{0}/{1}/{2}/{3}?useFallback={4}", searchParams);
+
+            //Get the HTTP response content from GET request
+            HttpContent httpContent = ReturnGetRespContent("Terms", searchParam);
+            if (httpContent != null)
+            {
+                Task<GlossaryTerm> term = ReadAsJsonAsync<GlossaryTerm>(httpContent);
+                var resultTerm = term.Result;
+                return resultTerm;
+            }
+            else
+            {
+                return null;
+            }
+            
+        }
+
+        /// <summary>
+        /// Gets the response content of a GET request.
+        /// </summary>
+        /// <param name="path">Path for client address</param>
+        /// <param name="param">Param in URL</param>
+        /// <returns>HTTP response content</returns>
+        public HttpContent ReturnGetRespContent(String path, String param)
+        {
+            HttpResponseMessage response = null;
+            HttpContent content = null;
+            String notFound = "NotFound";
+
+            
+            _client.DefaultRequestHeaders.Accept.Clear();
+            _client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+            // We want this to be synchronus, so call Result right away.
+            //NOTE: When using HttpClient.BaseAddress as we are, the path must not have a preceeding slash
+            response = _client.GetAsync(path + "/" + param).Result;
+            if (response.IsSuccessStatusCode)
+            {
+                content = response.Content;
+            }
+            else
+            {
+                string errorMessage = "Response: " + response.Content.ReadAsStringAsync().Result + "\nAPI path: " + this._client.BaseAddress.ToString() + path + "/" + param;
+                if (response.StatusCode.ToString() == notFound)
+                {
+                    // If trial is not found, log 404 message and return content as null
+                    log.Debug(errorMessage);
```

Comment:
It would be clearer if `content` were explicitly set to `null` here.

Reply:
The content variable is set to null before all of the API calls are made. But I can remove that part of the comment if it is confusing.

Reply:
On my earlier read-through, I struggled with the code which calls this routine. That may have colored my perceptions.

I don't have any good guidance to give on how to change the caller, so I think we're just gonna leave both as-is.